go/libraries/doltcore/schema/typeinfo/time.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2020 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package typeinfo
+
+import (
+	"fmt"
+
+	"github.com/src-d/go-mysql-server/sql"
+
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+// This is a dolt implementation of the MySQL type Time, thus most of the functionality
+// within is directly reliant on the go-mysql-server implementation.
+type timeType struct {
+	sqlTimeType sql.TimeType
+}
+
+var _ TypeInfo = (*timeType)(nil)
+
+var TimeType = &timeType{sql.Time}
+
+// ConvertNomsValueToValue implements TypeInfo interface.
+func (ti *timeType) ConvertNomsValueToValue(v types.Value) (interface{}, error) {
+	//TODO: expose the MySQL type's microsecond implementation and persist that to disk? Enables sorting
+	if val, ok := v.(types.String); ok {
```

Comment:
Again, not obvious this is the right choice. Much, much more expensive than using a time.Time under the hood.